### ISSUE TITLE
Tutorial required

### ISSUE THREAD

**Issue by JustinMinnaar** [EvergineTeam/Feedback#39]
The most pressing issue for me is that there is no clear step-by-step tutorial to get into using WaveEngine. Ideally, a tutorial needs to be created that will walk users through the basic concepts, show them how to write their own 'scripts' in C#, etc. If possible, perhaps I could get support from the developers of WaveEngine to write such a tutorial?

**Comment by hrkrx** [comment]
Is there a timeline when we can expect http://doc.waveengine.net/api3preview/ to be populated?

**Comment by horseyhorsey** [comment]
Updating all of these pages would be a good start. http://doc.waveengine.net/recipes/UI/Create-a-button.html

I found the `EntityManager` but there's no UI namespace anymore? 

Then you find it's recommended to use this other xaml rendering library and the old tutorial uses services for this but is nowhere to be found in the Editor.

**Comment by Lykdraft** [comment]
I don't know what the point is with this engine. 
No tutorial, no forum, no discord, no up-to-date API docs, no nothing...
Why even bother? The engine is around for a long time and this never was any different.
Chances are it never will be. Looks like a giant waste of time to me, to be honest.

**Comment by Hromon** [comment]
> I don't know what the point is with this engine.
> No tutorial, no forum, no discord, no up-to-date API docs, no nothing...
> Why even bother? The engine is around for a long time and this never was any different.
> Chances are it never will be. Looks like a giant waste of time to me, to be honest.

Name at least one worthy engine.

**Comment by Lykdraft** [comment]
> Name at least one worthy engine.


Well, I just can tell you with what I work(ed) and what I consider more "worthy" for myself.
(With working I mean actually finished games or at least running prototypes)

Cryengine, Lumberyard, Unreal. Unity, Godot, Flax, Stride, Armory3D, 
Korge had an experimental 3D branch. (not anymore as it seems since Korge Next, and language is Kotlin)
which actually was already easier and faster to work with than with this "thing", whatever it is
or wants to be. I also liked Babylon.JS (TS) andPlayCanvas (JS) and especially EZEngine (still in dev, but pretty cool already, nice TS API).  

Also the Chinese ones: The new Cocos Creator (still in preview, got RC1 here from the Chinese forums, release will be mid-February)  Egret and  Laya. Even there you have full API docs, Lol.

I would prefer **every-single-one-of them.** All of them have at least decent to full up-to-date API docs, forums, Discord or Gitter, written tuts and video tuts. Most of them have from at least a few to a lot of Github example projects to look at and dev examples for the most important aspects of the engine API (use-dases) as well...  Most of them have actual commercial releases under their belt as well and an actual **user-base.**  

So whatever Wave-Engine is (or wants to be) it is clearly not a game engine or an engine at all that's easy to work with.
But that's ok. I noticed after 2 hours, so not much time lost. Just don't get it, that's all. 

Wave Engine has **nothing** of the above. But still, you seem consider it the most "worthy" engine? I really wonder why.
It's not even ahead feature wise or quality wise with most of the above named engines. On the contrary. 
You consider an engine "the most worthy" where you don't even have at least up-to-date and thorough API docs?
Give me a break dude...

I am also not interested in your answer or will visit this engine or page or github or whatever anymore.
I am  just not interested in anything about it or anyone that tells me with a straight face this is worth anything at all.

Adios.

**Comment by Hromon** [comment]
> > Name at least one worthy engine.
> 
> Well, I just can tell you with what I work(ed) and what I consider more "worthy" for myself.
> _(With "worked with" I mean actually finished **at least** a small game or at minimum a running prototype with a full game loop and different game-states, can't give you a link to releases **yet**. At least one or two releases are coming this summer though...)_
> 
> So here it is:
> Cryengine, **Lumberyard**, Unreal. Unity, Godot, **Flax**, Stride, **Armory3D** & **Unigine**.
> Korge had an experimental 3D branch. (not anymore as it seems since Korge Next, and language is Kotlin)
> which actually was already easier and faster to work with than with this "thing", whatever it is
> or wants to be. I also liked Babylon.JS (TS) andPlayCanvas (JS) and especially **EZEngine** (still in dev, but pretty cool already, nice TS API).
> 
> Also the Chinese ones: The new **Cocos Creator** (still in preview, got RC1 here from the Chinese forums, release will be mid-February) Egret and Laya. Even there you have full API docs, Lol.
> 
> I also tried NeoAxis for a couple of days. Lovely, but this one is a bit of an empty desert and I failed to get a full prototype going. Still: I would prefer **every-single-one-of them.** All of them have at least decent to full up-to-date API docs, forums, Discord or Gitter, written tuts and video tuts. Most of them have from at least a few to a lot of Github example projects to look at and dev examples for the most important aspects of the engine API (use-dases) as well... Most of them have actual commercial releases under their belt as well and an actual **user-base.**
> 
> So whatever Wave-Engine is (or wants to be) it is clearly not a game engine or an engine at all that's easy to work with. But that's ok. I noticed after 2 hours, so not much time lost. I just don't get it, that's all.
> 
> Wave Engine has basically next to **nothing** of the above. But still, you seem to consider it the most "worthy" engine? I really wonder why. It's not even ahead feature-wise or quality-wise with most of the above-named engines. On the contrary. You consider an engine "the most worthy" where you don't even have at least up-to-date and thorough API docs? Care to tell me why that is?

I did not say that WaveEngine is worthy, I just asked you to list the engines that you could call them.

WaveEngine (3.x) has a cool architecture, but now it can hardly be called a full-fledged engine. It's more of a dev build. And just for the architecture I like the engine and I'm waiting for it to be released.

As for the engines you listed - half of them are frank feces.
Unity - broken crap that NuGet doesn't even support :/
Godot - too young and stupid as an engine, their node system is very slow and not very user-friendly.
Armory3D - hardly an engine at all, rather
student craft.

CRYENGINE - if I'm not mistaken, the normal documentation for it appeared not so long ago. И раньше в нём без ста грамм не разобраться)
Stride - I like it, but its architecture is not very good.

I did not work with the rest of the engines, I cannot say.

P.S. I use google translator, I hope you can figure out what I wrote: D

**Comment by Lykdraft** [comment]
All right no problem. :)

Most of the things you say are true. I also tagged the engines that I like the most **bold**.

Cryengine --> Correct, the full docs aren't up that long. Right now I am just waiting for the next release.
The engine seems to be pretty broken in some parts.

Unity --> Well, I hate Unity with a passion and everything that has to do with,  haha, so don't worry... 
I started with it, like probably a lot of people. At that time I didn't know what else is out there and
was happy when a cube moved over the screen.

Stride --> Yeah, I agree. 

Armory--> Unfortunately yes, since Lubos pretty much dropped it and focused on ArmorPaint there is not too much going on. It **is** evolving. But every slowly. Kind of like when Stride was dropped and left in community hands. A difficult position to be in.   like the guys that are still around in the ArmoryForums and Discord. though. They are working right now to get the upgrade to Blender 2.9x going.  I am still keeping a close eye on it.  Also, Blender integration isn't that big of a deal as quite some engines add tools for level blocking as well.  
Also, AFAIK Armory is in the process of being completely rewritten in C I think. 

Godot --> Haha, agreed. I used it, finished two small things there, and dropped it.
Can't stand the GDScript and how things are done there. Also, Godot just plain sucks with 3D.

Still, I prefer all of them to WE. Wouldn't work with them anymore, but still. I really wanted to like WaveEngine.
But there is just nowhere to go with it right now. This might change someday when there is more Material available.
Right now I just don't see any indie dev rolling with it. Like at all. I just don't see it.

And the translator works great, don't worry..;)
